src/dev/test_generator/verify/fixer.py

```
@@ -0,0 +1,185 @@
+                applied, backup_path = self._applier.apply_changes_with_backup(
+                    test_file, changes
+                )
+                if not applied:
+                    logger.warning("Fix attempt %d: apply failed for %s", i + 1, test_file.name)
+                    continue
+
+                # Verify the fix actually works by re-running the file.
+                # Exit code 0 = tests collected AND all passed — the only
+                # acceptable outcome.  Exit code 5 (no tests) means the AI
+                # deleted or corrupted all tests, which is NOT a fix.
+                proc = subprocess.run(
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)